tests/unit/test_routes.py

```diff
@@ -468,47 +511,75 @@ def test_create_resource(module_client, module_db, fake_auth_from_oc, fake_algol
 
 def test_update_resource(module_client, module_db, fake_auth_from_oc, fake_algolia_save):
     client = module_client
-
-    # Happy Path
     apikey = get_api_key(client)
 
-    response = client.put("/api/v1/resources/1",
-        json = dict(
-            name="New name",
-            languages=["New language"],
-            category="New Category",
-            url="https://new.url",
-            paid=False,
-            notes="New notes"
-        ),
-        headers = {'x-apikey': apikey}
-    )
-
+    # Happy Path
+    response = update_resource(client, apikey)
     assert (response.status_code == 200)
     assert (response.json['data'].get('name') == "New name")
 
     # Paid parameter as "FALSE" instead of False
-    response = client.put("/api/v1/resources/1",
-        json = dict(
-            name="New name 2",
-            languages=["New language"],
-            category="New Category",
-            url="https://new.url",
-            paid="FALSE",
-            notes="New notes"
-        ),
-        headers = {'x-apikey': apikey}
-    )
-
+    response = update_resource(client, apikey, name="New name 2", paid="FALSE")
     assert (response.status_code == 200)
     assert (response.json['data'].get('name') == "New name 2")
 
+    # Bogus Data
+    name = False
+    url = "htt://bad_url.doesnotexist"
+    category = True
+    languages = False
+    paid = "Bad Data"
+    notes = True
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 422)
+
+    # A String to Big for the DB
+    long_string = "x" * 6501
+    name = long_string
+    url = long_string
+    category = long_string
+    languages = long_string
+    paid = True
+    notes = long_string
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 422)
+
+    # Unicode Characters
+    name = "😀"
+    url = None
+    category = None
+    languages = None
+    paid = True
+    notes = "∞"
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 200)
```

Comment:
Note that this one is fine because you don't _have_ to include anything in particular on an update.